### PR TITLE
[flink] Support nullability change when altering table on flink 1.17

### LINKED ISSUE #1111: [Feature] Support nullability change when altering table on flink 1.17
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Support nullability change when altering table on flink 1.17

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by schnappi17**
fix #1111 

Flink is able to modify nullable column to not null column by setting `table.exec.sink.not-null-enforcer` to `DROP` to dropping those null-value records with not-null column silently.

### CODE REVIEW

**Comment by yuzelin** [comment]
Please solve check-style violations.

**Comment by schnappi17** [comment]
> Please solve check-style violations.

@yuzelin Thanks for reminding, updated it.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 1]
```diff
@@ -357,7 +374,9 @@ public void alterTable(
         List<SchemaChange> changes = new ArrayList<>();
         if (null != tableChanges) {
             List<SchemaChange> schemaChanges =
-                    tableChanges.stream().map(this::toSchemaChange).collect(Collectors.toList());
+                    tableChanges.stream()
+                            .flatMap(tableChange -> (toSchemaChange(tableChange).stream()))
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:378]: `(toSchemaChange(tableChange).stream())`
The `()` is redundant.

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/SchemaChangeITCase.java [hunk 2]
```diff
@@ -293,10 +304,40 @@ public void testModifyNullability() {
...
+        // Insert a null value
+        sql("INSERT INTO T VALUES('aaa', 'bbb', 'ccc', 1, CAST(NULL AS FLOAT))");
+        result = sql("select * from T");
+        assertThat(result.toString()).isEqualTo("[+I[aaa, bbb, ccc, 1, null]]");
+
+        // Then nullable -> not null
+        tEnv.getConfig()
+                .set(
+                        ExecutionConfigOptions.TABLE_EXEC_SINK_NOT_NULL_ENFORCER,
+                        ExecutionConfigOptions.NotNullEnforcer.DROP);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/SchemaChangeITCase.java:336]: This is important message. Currently, the doc `altering table - Changing Column Nullability` only record how to change `not null` to `nullable`. Please also document this usage for flink.

**Review by yuzelin — COMMENTED** [review]
Thanks for contributing. I left some comments, please take a look.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java (hunk 1 above)
Comment (schnappi17) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java]: Thanks a lot, updated.

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/SchemaChangeITCase.java (hunk 2 above)
Comment (schnappi17) [reply paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/SchemaChangeITCase.java:336]: Sure, documented it.

**Review by yuzelin — APPROVED** [review]
LGTM

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 3]
```diff
@@ -249,44 +249,61 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
...
+            schemaChanges.add(
+                    SchemaChange.renameColumn(
+                            modify.getOldColumnName(), modify.getNewColumnName()));
+            return schemaChanges;
         } else if (change instanceof ModifyPhysicalColumnType) {
             ModifyPhysicalColumnType modify = (ModifyPhysicalColumnType) change;
-            return SchemaChange.updateColumnType(
-                    modify.getOldColumn().getName(),
-                    LogicalTypeConversion.toDataType(modify.getNewType().getLogicalType()));
+            schemaChanges.add(
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:278]: If nullable change?

**Pull request merged**